src/psrsearch.jl

```diff
@@ -38,26 +36,25 @@ julia> z_n_binned([10., 0., 0., 0., 0.], 2)
 40.0
 ```
 """
-function z_n_binned(profile::Array{T,1}, n::Integer)::T where {T<:AbstractFloat}
+function z_n_binned(profile::AbstractVector{T}, n::Integer)::T where {T<:AbstractFloat}
     total = sum(profile)
     N = length(profile)
-    phase = Array(range(0, stop = N - 1) / N) * 2 * pi
+    phase = range(0, stop = N - 1) / N * 2 * pi
 
-    if total == 0
-        return 0
+    if iszero(total)
+        return zero(T)
```

Comment:
This is for [type-stability](https://docs.julialang.org/en/v1/manual/performance-tips/#Write-%22type-stable%22-functions): you ensure you return a zero always of the same type (even though the type annotation `::T` to the method has the same effect)